pkgs/tools/networking/oneshot/default.nix

```
@@ -0,0 +1,26 @@
+{ lib, fetchFromGitHub, buildGoModule }:
+
+buildGoModule rec {
+  pname = "oneshot";
+  version = "1.0.1";
+
+  src = fetchFromGitHub {
+    owner = "raphaelreyna";
+    repo = "oneshot";
+    rev = "v${version}";
+    sha256 = "11v6ljp645hp0fh16s1azxxj6644hrq8j1varwycapyyn0661azs";
+  };
+
+  goPackagePath = "github.com/raphaelreyna/oneshot";
+  vendorSha256 = "02y4pblsl6f069hm521f8pzdrhkgrs5qh49zj6xc3bjm6av1pamm";
+
+  subPackages = [ "." ];
+
+  meta = with lib; {
+    description = "A first-come-first-serve single-fire HTTP server";
+    homepage = https://github.com/raphaelreyna/oneshot;
+    license = licenses.mit;
+    maintainers = with maintainers; [ edibopp ];
+    platforms = platforms.linux;
```

```suggestion
    platforms = platforms.all;
```

Or is there a good reason to believe that this won't build on macOS?

---

No, I am pretty sure it works on MacOS, as Go apps don't really have much platform-dependent stuff in them. But I don't have any Apple device to test that assumption and thus cautiously left it out. Fine by me to include it despite that.